### DISCUSSION TITLE
Forward peerDependencies

### DISCUSSION THREAD

**Discussion by duarten** [pnpm/pnpm#4977] [Q&A] [answered]
👋🏻 Consider the following `package.json`:

```json
{
    "name": "foo",
    "private": true,
    "type": "module",
    "dependencies": {
        "mydep": "1.0.0"
    },
    "peerDependencies": {
        "mydep-peer": "^1.0.0"
    }
}
```

There's a direct dependency on `mydep`, which in turns requires a peer dependency on `mydep-peer`. However, my package also declares a peer depedency on `mydep-peer`.

In this scenario I would expect pnpm to understand that `mydep`'s peer dependency is covered by the package's own peer dependency. However, pnpm still warns the peer dependency isn't satisfied.

Am I wrong in expecting this? Or is it a known limitation?

**Comment by robations** [comment, marked as answer]
@duarten I'm often confused by peer deps, but I think you might also need to specify `mydep-peer` elsewhere:

- if `foo` is published/consumed by another package, add it to `devDependencies` (in foo)
- if `foo` is used in isolation, remove it from `peerDependencies`, and add to `dependencies` (in foo)

Give this a shot, maybe someone can correct me if wrong! Also check out the `auto-install-peers` if you haven't seen it.

**Reply by duarten** [reply]
I think adding the peer as a `devDependency` is a good solution :)
